### PR TITLE
-Author:Hunan Khachatryan: Junit testing.

### PR DESCRIPTION
**Pull request by HunanKhachatryan**
Calculator for calculate the expression from file and write result in file.

### CODE REVIEW

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Stack;
+
+public class Calculate {
+    public static  void main(String[] args) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        String filename = "/home/hunan/IdeaProjects/Calculate/src/main/java/file.txt";
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: hardcoded path

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            System.err.format("IOException: %s%n", e);
+        }
+
+
+        System.out.println(sb);
+
+        String[] tmp = sb.toString().split("=");
+
+        String leftPart = tmp[0];
+        Calculate  expression = new Calculate();
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: use self instead of this

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                System.out.println(result);
+
+            }
+        }else {
+            System.out.println("In expression the paranthes is not balanced");
+
+        }
+    }
+    public  void writeFile(String filename, Double text){
+        try{
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: use try-with-resources

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+        }catch(Exception e){
+            System.out.println(e);
+        }
+    }
+
+
+    private int priority(char op){
+
+        if(op == '+'||op == '-')
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: switch instead of the if-s

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            case '-': return a - b;
+            case '*': return a * b;
+            case '/': return a / b;
+            default: return 0;
+        }
+
+    }
+
+
+    public boolean AreParanthesisBalanced(String str)
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: A function name should start with a small letter

File: Hunan_Khachatryan/JavaCalculator/Calculate.java (hunk 2 above)
Comment (HunanKhachatryan) [reply Hunan_Khachatryan/JavaCalculator/Calculate.java]: Sorry but i dont understand your note.

File: Hunan_Khachatryan/JavaCalculator/Calculate.java (hunk 1 above)
Comment (HunanKhachatryan) [reply Hunan_Khachatryan/JavaCalculator/Calculate.java]: Is fixed.

File: Hunan_Khachatryan/JavaCalculator/Calculate.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+            }
+        } else {
+            System.out.println("In expression the paranthes is not balanced");
+
+        }
+    }
+
+    public void writeInFile(String filename, Double result) throws IOException {
+        try (FileWriter fWrite = new FileWriter(filename, true)) {
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/Calculate.java]: Catch the exception in the scope of this function.

File: Hunan_Khachatryan/JavaCalculator/my_app/src/main/java/am/Instigate/app/App.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+            } else {
+                double result = this.evalExpression(leftPart);
+                String text = " = " + result;
+                this.writeInFile(filename, text);
+
+            }
+
+        } else {
+            throw new IOException("In expression the paranthes is not balanced");
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/my_app/src/main/java/am/Instigate/app/App.java]: why IOException exception

File: Hunan_Khachatryan/JavaCalculator/my_app/src/test/java/am/Instigate/app/AppTest.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest {
+    /**
+     * Rigorous Test :-)
+     */
+    @Rule
+    public ExpectedException exeption = ExpectedException.none();
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/my_app/src/test/java/am/Instigate/app/AppTest.java]: misspelling

File: Hunan_Khachatryan/JavaCalculator/my_app/src/test/java/am/Instigate/app/AppTest.java [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+        assertEquals("Evaluate expression is false ",60.0, test.evalExpression("( 10 + 5 ) * 4"), 0.0);
+
+    }
+
+    //test that checks the expression calculation function
+    @Test
+    public void evalExpressionFalse() {
+        App test = new App();
+        assertNotEquals("Evaluate expression is true ",15.0, test.evalExpression("10 + 5 * 4"), 0.0);
+
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/my_app/src/test/java/am/Instigate/app/AppTest.java]: what are you checking with this case?

File: Hunan_Khachatryan/JavaCalculator/my_app/src/main/java/am/Instigate/app/App.java [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Stack;
+
+/**
+ * Hello world!
+ */
+
```
Comment (taronqalashyan) [inline Hunan_Khachatryan/JavaCalculator/my_app/src/main/java/am/Instigate/app/App.java]: ?

**Pull request closed without merging**